waku/incentivization/txid_proof.nim

```diff
@@ -0,0 +1,62 @@
+import std/options, chronos, web3, stew/byteutils, stint, strutils
+
+import waku/incentivization/rpc
+
+
+# Function to convert a hex string to Address
+proc toAddress*(hexStr: string): Address =
+  # Remove the "0x" prefix if it exists
+  let cleaned = if hexStr.startsWith("0x"): hexStr[2..^1] else: hexStr
+
+  # Ensure the length is exactly 40 characters (20 bytes)
+  if cleaned.len != 40:
+    raise newException(ValueError, "Invalid hexadecimal string length for Address")
+
+  var arr: array[20, byte]
+  for i in 0 ..< 20:
+    let byteValue = cleaned[i * 2 ..< i * 2 + 2]  # Get two hex characters
+    arr[i] = byte(parseHexInt(byteValue))
+
+  result = Address(arr)
```

Comment:
We should avoid the use of the implicit `result` attribute. It is generally better to explicitly return a value. See the suggestion considering that this proc return a `Result[Address, string]`
```suggestion
  return ok(Address(arr))
```